### DISCUSSION TITLE
PROCEDUREs aren't shown in the Functions view

### DISCUSSION THREAD

**Discussion by ChristianSi** [supabase/supabase#33007] [Feature Requests]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Occasionally it's necessary to create a PROCEDURE instead of a FUNCTION. Specifically, PROCEDUREs invoked with CALL can do their own transaction management, while FUNCTIONs cannot.

It's possible to create, replace, and drop PROCEDUREs using the SQL editor, but they don't show up in the Functions view, which makes accessing and editing them very difficult.

## To Reproduce

1. Use the SQL editor to create a PROCEDURE, for example:

```sql
CREATE OR REPLACE PROCEDURE public.myproc()
 LANGUAGE plpgsql
AS $procedure$
declare
begin
  -- do something here
end;$procedure$
;
```

2. Open the Database / Functions view and select the specified schema (if necessary) to find it. It won't be there, and if search for it using the search box, it won't show up either.

## Expected behavior

PROCEDUREs should be shown in the Functions view together with actual FUNCTIONs. There should be some symbol or similar that allows to distinguish which is which.

Alternatively, there could be a separate Database / Procedures view just for them.

Ideally it should also be possible to create them using the GUI. Either by adding to the "Add a new function" view a checkbox "Create as procedure" (or something similar). Or, if there is a new Database / Procedures view, by adding a "Add a new procedure" button there.

## System information

All/latest.

**Comment by Hallidayo** [comment]
Hi everyone, due to inactivity on this issue I've moved the issue over to discussions/enhancements.

**Comment by osamakawish** [comment]
Also having this bug. I think it can be worked around with a void returning function though. So it's not entirely a serious bug. But it can be annoying if you're new to it.

Still, quite sure you can't rollback with functions, so it is an annoying workaround.

**Reply by ChristianSi** [reply]
No, functions cannot generally replace procedures, especially since only procedures can do their own transaction management.

**Comment by TSIA-SN** [comment]
We have also run into this issue along with the fact that Procedures aren't picked up by the TypeScript type generator, we need to use procedures due to their support for `COMMIT` statements within the body, which functions do not support.
